### PR TITLE
#153 Use git to get unstable semver

### LINKED ISSUE #153: Add githash to Semvers for unstable docker images from master
When PR's are merged to master push Docker images with tag following the pattern
```
<tagged-release>-<commit-count>-<githash>
```

This can be generated using:
```
git describe --tags
```

### PR DESCRIPTION
**Pull request by whatnick**
Closes #153 
- Use git to fetch unstable semver
- Push unstable Semver to dockerhub
- Scan the latest unstable semver with Trivy instead of latest
- Scan release semver with Trivy ( more scanning more security ? )

### CODE REVIEW

**Review by jeremyh — APPROVED** [review]

File: .github/workflows/docker.yml [hunk 1]
```diff
@@ -35,6 +42,15 @@ jobs:
           password: "${{ secrets.DockerPassword }}"
           build_extra_args: "--build-arg=ENVIRONMENT=deployment"
 
+      - name: Build and Push unstable Docker image tag
+        uses: whoan/docker-build-with-cache-action@v4
+        with:
+          image_name: ${{ env.IMAGE_NAME }}
+          username: gadockersvc
+          image_tag: ${{ env.UNSTABLE_TAG }}
```
Comment (alexgleith) [inline .github/workflows/docker.yml]: We want to push `latest` here too

File: .github/workflows/docker.yml [hunk 2]
```diff
@@ -35,6 +42,15 @@ jobs:
           password: "${{ secrets.DockerPassword }}"
           build_extra_args: "--build-arg=ENVIRONMENT=deployment"
 
+      - name: Build and Push unstable Docker image tag
+        uses: whoan/docker-build-with-cache-action@v4
```
Comment (alexgleith) [inline .github/workflows/docker.yml]: I think we want `if != 'release'`

File: .github/workflows/docker.yml [hunk 3]
```diff
@@ -35,6 +42,15 @@ jobs:
...
+        with:
+          image_name: ${{ env.IMAGE_NAME }}
+          username: gadockersvc
+          image_tag: ${{ env.UNSTABLE_TAG }}
+          password: "${{ secrets.DockerPassword }}"
+          build_extra_args: "--build-arg=ENVIRONMENT=deployment"
+
       - name: Build and Push latest Docker image (old name)
```
Comment (alexgleith) [inline .github/workflows/docker.yml]: Let's get rid of the old docker image... moving forwards!

File: .github/workflows/docker.yml (hunk 1 above)
Comment (whatnick) [reply .github/workflows/docker.yml]: Latest is the default. Pushed in another step. Not sure if the action supports multiple tags. Looking at it in a bit.

Comment (whatnick) [reply .github/workflows/docker.yml]:
```
image_tag: Tag(s) of the image to build. Allows multiple comma-separated tags (e.g. one,another) (default: latest).
```
Squashing into one step.

File: .github/workflows/docker.yml (hunk 3 above)
Comment (whatnick) [reply .github/workflows/docker.yml]: Too many changes I am not sure I have changed them all in all my environments. Delayed deprecation.

**Comment by jeremyh** [comment]
I shouldn’t review parts of the system I have little experience with 🙂

**Comment by whatnick** [comment]
> I shouldn’t review parts of the system I have little experience with 🙂

Inducting / Subverting you into the ways of the docker. But just keeping you aware of the changes we are making. Need that increment change tested and shipped out soon to keep live dataflows happy. Pushing to dockerhub is sort of the first step to that.

**Review by alexgleith — APPROVED** [review]

**Pull request merged**